client/state/sites/hooks/use-site-global-styles-status.ts

```diff
@@ -33,13 +68,38 @@ export const getGlobalStylesInfoForSite = ( siteId: number | null ): GlobalStyle
 		.catch( () => DEFAULT_GLOBAL_STYLES_INFO );
 };
 
-export function useSiteGlobalStylesStatus( siteId: number ): GlobalStylesStatus {
-	const { data } = useQuery( {
-		queryKey: [ 'globalStylesInfo', siteId ],
-		queryFn: () => getGlobalStylesInfoForSite( siteId ),
+export function useSiteGlobalStylesStatus(
+	siteIdOrSlug: number | string | null = null
+): GlobalStylesStatus {
+	const selectedSiteId = useSelector( getSelectedSiteId );
+
+	// When site id is null it means that the site hasn't been created yet.
+	const siteId = useSelector( ( state ) => {
+		const currentSiteId = siteIdOrSlug ?? selectedSiteId;
+		if ( ! currentSiteId ) {
+			return null;
+		}
+
+		const site = getSite( state, currentSiteId );
+		return site?.ID ?? null;
+	} );
+
+	const { data: globalStylesOnPersonalExperimentAssignment } = useQuery( {
+		queryKey: [ 'globalStylesOnPersonalExperimentAssignment', siteId ],
+		queryFn: () => getExperimentAssignment( 'calypso_global_styles_personal' ),
+		placeholderData: null,
+		refetchOnWindowFocus: false,
+		enabled: typeof window !== undefined && siteId === null,
```

Comment:
Is this what Jessie mentioned about checking window in SSR context?

Reply:
Yes and no. The ExPlat library shouldn't be used in SSR contexts, but we're not using the ExPlat library anymore. However, we don't need to run the experiment in SSR contexts, which is why we avoid making the API call in such cases.